core/arch/decode.h

```diff
@@ -329,6 +329,12 @@ enum {
     OPSZ_32_of_64,        /**< 256 bits: half of ZMM. */
     OPSZ_4_of_32_evex64,  /**< 32 bits: can be part of YMM or ZMM register. */
     OPSZ_8_of_32_evex64,  /**< 64 bits: can be part of YMM or ZMM register. */
+    OPSZ_4_dimode8,  /**< 4 or 8 bytes, but not based on rex prefix, instead dependent
+                      * on 32-bit/64-bit mode.
+                      */
+    OPSZ_8_dimode16, /**< 8 or 16 bytes, but not based on rex prefix, instead dependent
```

Comment:
Please follow the existing naming conventions: OPSZ_8x16

Reply:
In addition, the size resolution belongs into resolve_var_x64_size. Fixed.